pkg/controller.v1/pytorch/pytorchjob_controller.go

```diff
@@ -215,6 +218,51 @@ func (r *PyTorchJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 
+	// inject watching for job related service
+	if err = c.Watch(&source.Kind{Type: &corev1.Service{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &kubeflowv1.PyTorchJob{},
+	}, predicate.Funcs{
+		CreateFunc: util.OnDependentCreateFunc(r.Expectations),
+		UpdateFunc: util.OnDependentUpdateFunc(&r.JobController),
+		DeleteFunc: util.OnDependentDeleteFunc(r.Expectations),
+	}); err != nil {
+		return err
+	}
+
+	// inject watching for job related objects,such as PodGroup
+	if config.Config.WatchedResources != nil {
```

Comment:
Can you rebase your code and add it for all frameworks? Also, the volcano static references and watches in the current code should be removed.

Reply:
When I wrote this feature, I did not realize that it was already supported and merged. @_@